### ISSUE TITLE
Unexpected behavior with the populate method leading to memory issues

### ISSUE THREAD

**Issue by yassine-cc** [Automattic/mongoose#12834]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.8.1

### Node.js version

18.12.1

### MongoDB server version

irrelevant to this issue

### Description

When trying to use the populate static method (and i suspect the instance method as well) with a match on _id, mongoose completely ignores the ids it is trying to match on in its query, which can lead to a huge and unexpected query results size and issues with the garbage collector.

we've came across this problem when our backend had a sudden problem with memory usage and the garbage collector constantly blocking the event loop.

The code causing this issue is in https://github.com/Automattic/mongoose/blob/master/lib/helpers/populate/createPopulateQueryFilter.js line 13, it doesn't  add the internal match on `_id` if  one is already provided instead of combining the two

### Steps to Reproduce

```typescript

mongoose.set('debug', true);

const personSchema = new Schema({
  name: String,
  age: Number,
  stories: [{ type: Schema.Types.ObjectId, ref: 'Story' }],
});

const storySchema = new Schema({
  author: { type: Schema.Types.ObjectId, ref: 'Person' },
  title: String,
  fans: [{ type: Schema.Types.ObjectId, ref: 'Person' }],
});

const Story = mongoose.model('Story', storySchema);
const Person = mongoose.model('Person', personSchema);

function objectIdWithTimestamp(timestamp) {
    if (typeof(timestamp) == 'string') {
        timestamp = new Date(timestamp);
    }
    var hexSeconds = Math.floor(timestamp/1000).toString(16);
   return ObjectId(hexSeconds + "0000000000000000");
}

async function test() {
  const person = await Person.findOne({ });

  await Story.populate(person, {
    path: 'stories',
    match: {
      _id: { $gte: objectIdWithTimestamp(new Date('2020-01-01')) },
    },
  });
}

mongoose
  .connect('mongodb://127.0.0.1:27017/db1')
  .then(test)
  .then(() => mongoose.disconnect());  
```
In this example, the query that would be sent to the database would have only the match on the Id that was provided by the code above, which could return a huge portion of the documents in the affected collection

Here are the printed queries to the console:
```javascript
people.findOne({}, { projection: {} })
stories.find({ _id: { '$gte': new ObjectId("5e0be1000000000000000000") } }, { skip: undefined, limit: undefined, perDocumentLimit: undefined, projection: {}})
```

### Expected Behavior

The expected behavior would be that whatever match conditions are provided on the `_id`, would be combined using `$and`  with the internal match that the populate method would normally produce  `{ _id: {$in: [ /* Array of object ids*/]}}` and spit out a query like this:
```javascript
stories.find(
  {
    $and: [
      { _id: { $gte: new ObjectId('5e0be1000000000000000000') } },
      { _id: { $in: [/* Array of object ids*/] } },
    ],
  },
  {
    skip: undefined,
    limit: undefined,
    perDocumentLimit: undefined,
    projection: {},
  }
);
```

**Comment by IslandRhythms** [comment]
```javascript
const mongoose = require('mongoose');

const {Schema} = mongoose;

mongoose.set('debug', true);

const personSchema = new Schema({
  name: String,
  age: Number,
  stories: [{ type: Schema.Types.ObjectId, ref: 'Story' }],
});

const storySchema = new Schema({
  author: { type: Schema.Types.ObjectId, ref: 'Person' },
  title: String,
  fans: [{ type: Schema.Types.ObjectId, ref: 'Person' }],
});

const Story = mongoose.model('Story', storySchema);
const Person = mongoose.model('Person', personSchema);

function objectIdWithTimestamp(timestamp) {
    if (typeof(timestamp) == 'string') {
        timestamp = new Date(timestamp);
    }
    var hexSeconds = Math.floor(timestamp/1000).toString(16);
    const id = mongoose.Types.ObjectId(hexSeconds + "0000000000000000");
    console.log('the genereated id', id);
    return id;
}

async function test() {
  await mongoose.connect('mongodb://localhost:27017');
  await mongoose.connection.dropDatabase();
  const entry = await Person.create({
    name: 'Test',
    age: 1,
    stories: []
  });
  const doc = await Story.create({
    author: entry._id,
    title: 'Test title',
    fans: entry._id
  });

  const doc2 = await Story.create({
    author: entry._id,
    title: 'Another test title',
    fans: entry._id
  });

  const entry2 = await Person.create({
    name: 'Test2',
    age: 2,
    stories: [doc._id, doc2._id]
  });
  const person = await Person.findOne({_id: entry2._id});
  console.log('The retrieved person', person)
  const story = await Story.populate(person, {
    path: 'stories',
    match: {
      _id: { $gte: objectIdWithTimestamp(new Date('2023-02-02')) },
    },
  });
  console.log('The story', story)
  console.log(await Person.find());
}

test();
```

I'm having a hard time understanding what the issue is. When you call populate, it is populating the indicated docs by your `$gte` condition. If you change the date to the future, nothing is populated. 
In your expected behavior section, you say the query should look like

```javascript
stories.find(
  {
    $and: [
      { _id: { $gte: new ObjectId('5e0be1000000000000000000') } },
      { _id: { $in: [/* Array of object ids*/] } },
    ],
  },
  {
    skip: undefined,
    limit: undefined,
    perDocumentLimit: undefined,
    projection: {},
  }
);
```

But what specific objectIds are you expecting in the `$in` condition that wouldn't be covered by the `$gte`?

**Comment by yassine-cc** [comment]
Since we are using populate to retrieve the stories related to the person object we are currently processing, I expect that mongoose would match only stories related to that specific person document, regardless of whether I provided an additional match condition on the `_id` field.

Let's say we have a document Person holding 10 `storyIds` in its `stories` field, and we only want to poplate the  Person document's stories that were created in the last 12 months.

Right now with things as is, since we provided a match option on `_id`, the produced query would fetch **all** stories of the past year  (potentially millions of records) which is a problem.

Naturally, it wouldn't populate the person record with all of them as not all of them are present in the `stories` field, however the following things happen:

- Querying the database returning millions of records 
- Loading them  into memory
- Potentially instanciating them in some class
- Filtering them locally

Which has a huge potential performance impact, we've encountred this same issue in production where an infrequent query (once every few minutes generally) would happen, and the server went unresponsive for 10 seconds to a minute at a time.
